### PR TITLE
roles: print matching lines from journal

### LINKED ISSUE #253: journal_search task should also print out offending line
If we fail because we found e.g. `avc: denied` in the journal, we should also print out the line from the journal which caused the failure. This would make debugging it easier.

### PR DESCRIPTION
**Pull request by miabbott**
Previously, when the `journal_search` role would fail, it would only
print a message saying that the matching line had been found.  This is
only partially useful if you want to see the actual matching line in
the test output.

This changes the `journal_search` role to crudely print the matching
lines by way of running a `grep` against the output of `journalctl`
after a match has been found.  The reason for doing this operation
after the match has been found is because the command `journal | grep
<string>` may end up in the journal if it is run with elevated
privileges via Ansible (i.e. using `sudo`)

Closes #253

### CODE REVIEW

File: roles/journal_search/tasks/main.yml [hunk 1]
```diff
@@ -31,16 +31,33 @@
...
 - name: Search for string in the journal
   fail:
     msg: "Unable to find string '{{ search_string }}' in the journal"
   when:
     - not fif
-    - search_string not in j.stdout
+    - not found_string
+
+# We scrape the journal manually after the fact because the contents of the
+# 'shell:' command below will end up in the journal when it is run on the
```
[the remark this answers is not in the document]
Comment (miabbott) [reply roles/journal_search/tasks/main.yml]: Hmm, in my own tests adding the `_TRANSPORT=audit` expression may be useful.  I'm going to pursue this a little further.

[the remark this answers is not in the document]
Comment (miabbott) [reply roles/journal_search/tasks/main.yml]: For the case of AVC denials, using `_TRANSPORT=audit` would be very useful, but this role is intended to be a generic way of looking for a string in the journal, so we can't use that all the time.

We could probably still conditionally add that type of filter via a passed in arg, though...

**Comment by miabbott** [comment]
This should be ready for review again.  The scope grew a bit, but I think the end result was role that can be re-used with more flexibility.

**Comment by miabbott** [comment]
Regarding the test failures:

- centos/7/atomic is #254 

- fedora/26/atomic looks like `/var/mnt/lost+found` is ending up as `unlabeled_t`;  I haven't been able to reproduce this locally.  Will talk to @jlebon about the VMs being provisioned for PAPR

**Comment by jlebon** [comment]
> fedora/26/atomic looks like `/var/mnt/lost+found` is ending up as `unlabeled_t`

Hmm, not sure what's going on there. I don't see it locally, nor in a manually provisioned OpenStack node either. But I do see it in PAPR nodes. Seems like `/var/mnt` itself is labeled `unlabeled_t`.

**Comment by jlebon** [comment]
https://github.com/projectatomic/papr/pull/68

**Comment by miabbott** [comment]
[(3) improve output from 'journal_search' role in [a-h-t]](https://trello.com/c/uw9yTJJE/852-3-improve-output-from-journalsearch-role-in-a-h-t)

**Comment by miabbott** [comment]
- centos/7/atomic:  this is #254, which should be fixed with #261.  We can wait until that lands and try retesting this context.

**Comment by mike-nguyen** [comment]
bot, retest this please

**Comment by miabbott** [comment]
@mike-nguyen We good to merge this?

**Comment by mike-nguyen** [comment]
Ready to go!

**Pull request merged**